timeAsyncMethod.coffee

```diff
@@ -0,0 +1,30 @@
+
+module.exports = (obj, methodName, prefix, logger) ->
+	metrics = require('./metrics')
+
+	if typeof obj[methodName] != 'function'
+		throw new Error("[Metrics] expected object property '#{methodName}' to be a function")
+
+	realMethod = obj[methodName]
+	key = "#{prefix}.#{methodName}"
+
+	obj[methodName] = (originalArgs...) ->
+
+		[firstArgs..., callback] = originalArgs
+		if !callback? || typeof callback != 'function'
+			throw new Error(
```

Comment:
Can we just fail silently here, or log an error. Throwing is a bit harsh and could crash the process on a non-fatal error. We sometimes fire off a method in a 'fire and forget way' with no callback. Need to make sure to just call the method with the supplied args if so.